sdk/storage/Azure.Storage.DataMovement.Blobs/tests/AppendBlobDirectoryStartTransferDownloadTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+extern alias DMBlobs;
+extern alias BaseBlobs;
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using BaseBlobs::Azure.Storage.Blobs;
+using BaseBlobs::Azure.Storage.Blobs.Models;
+using BaseBlobs::Azure.Storage.Blobs.Specialized;
+
+namespace Azure.Storage.DataMovement.Blobs.Tests
+{
+    [DataMovementBlobsClientTestFixture]
+    public class AppendBlobDirectoryStartTransferDownloadTests
+        : BlobDirectoryStartTransferDownloadTestBase
+    {
+        public AppendBlobDirectoryStartTransferDownloadTests(
+            bool async,
+            BlobClientOptions.ServiceVersion serviceVersion)
+        : base(async, serviceVersion)
+        {
+        }
+
+        protected override async Task CreateBlobClientAsync(
+            BlobContainerClient container, string blobName, Stream contents, CancellationToken cancellationToken = default)
+        {
+            AppendBlobClient appendBlobClient = container.GetAppendBlobClient(blobName);
+            await appendBlobClient.CreateIfNotExistsAsync();
+            if (contents != default)
+            {
+                if (contents.Length == 0)
+                {
+                    // cannot append a stream that has a Content-Length of 0 so we adjust to 1
```

Comment:
If we're intending to create an empty append blob, then this code path would guarantee that could never happen.

For this case let's just skip the `AppendBlock` step. That way there's an empty append blob

Reply:
Good idea. I will modify this for PageBlobDirectoryStartTransferDownloadTests as well